cartographer_ros/cartographer_ros/node.cc

```diff
@@ -466,51 +466,53 @@ bool Node::ValidateTopicNames(
+      {TrajectoryState::FINISHED, "FINISHED"},
+      {TrajectoryState::FROZEN, "FROZEN"},
+      {TrajectoryState::DELETED, "DELETED"},
+  };
+  status_response.message =
+      absl::StrCat("Trajectory ", trajectory_id, " is in '",
+                   state_to_string.at(trajectory_state), "' state.");
+  if (valid_states.count(trajectory_state)) {
+    status_response.code = cartographer_ros_msgs::StatusCode::OK;
+  } else {
+    status_response.code = cartographer_ros_msgs::StatusCode::INVALID_ARGUMENT;
+  }
+  return status_response;
+}
+
 cartographer_ros_msgs::StatusResponse Node::FinishTrajectoryUnderLock(
     const int trajectory_id) {
-  auto trajectory_states = map_builder_bridge_.GetTrajectoryStates();
-
   cartographer_ros_msgs::StatusResponse status_response;
   if (trajectories_scheduled_for_finish_.count(trajectory_id)) {
-    const std::string message = absl::StrCat("Trajectory ", trajectory_id,
-                                             " already pending to finish.");
+    status_response.message = absl::StrCat("Trajectory ", trajectory_id,
+                                           " already pending to finish.");
     status_response.code = cartographer_ros_msgs::StatusCode::OK;
-    status_response.message = message;
-    LOG(INFO) << message;
+    LOG(INFO) << status_response.message;
     return status_response;
   }
 
   // First, check if we can actually finish the trajectory.
-  if (!(trajectory_states.count(trajectory_id))) {
-    const std::string error =
-        absl::StrCat("Trajectory ", trajectory_id, " doesn't exist.");
-    LOG(ERROR) << error;
-    status_response.code = cartographer_ros_msgs::StatusCode::NOT_FOUND;
-    status_response.message = error;
-    return status_response;
-  } else if (trajectory_states.at(trajectory_id) == TrajectoryState::FROZEN) {
-    const std::string error =
-        absl::StrCat("Trajectory ", trajectory_id, " is frozen.");
-    LOG(ERROR) << error;
-    status_response.code = cartographer_ros_msgs::StatusCode::INVALID_ARGUMENT;
-    status_response.message = error;
-    return status_response;
-  } else if (trajectory_states.at(trajectory_id) == TrajectoryState::FINISHED) {
-    const std::string error = absl::StrCat("Trajectory ", trajectory_id,
-                                           " has already been finished.");
-    LOG(ERROR) << error;
-    status_response.code =
-        cartographer_ros_msgs::StatusCode::RESOURCE_EXHAUSTED;
-    status_response.message = error;
-    return status_response;
-  } else if (trajectory_states.at(trajectory_id) == TrajectoryState::DELETED) {
-    const std::string error =
-        absl::StrCat("Trajectory ", trajectory_id, " has been deleted.");
-    LOG(ERROR) << error;
-    status_response.code =
-        cartographer_ros_msgs::StatusCode::RESOURCE_EXHAUSTED;
-    status_response.message = error;
-    return status_response;
+  cartographer_ros_msgs::StatusResponse check_response = CheckTrajectoryState(
```

Comment:
naming:response

Reply:
Changed it a bit differently: we can fill the already existing `status_response` variable